Sources/PointFree/Pricing.swift

```diff
@@ -25,90 +25,64 @@ public enum Pricing: Codable, DerivePartialIsos {
     case yearly
   }
 
-  private enum CodingKeys: String, CodingKey {
+  enum Lane: String, Codable {
     case individual
     case team
-    case lane
   }
 
-  private enum Lane: String, Codable {
-    case individual
-    case team
-  }
-
-  public init(from decoder: Decoder) throws {
-    let container = try decoder.container(keyedBy: CodingKeys.self)
-    let billing = try container.decodeIfPresent(Billing.self, forKey: .individual)
-    let quantity = try container.decodeIfPresent(Int.self, forKey: .team)
-    let lane = try container.decodeIfPresent(Lane.self, forKey: .lane)
-
-    if let lane = lane, let billing = billing, let quantity = quantity {
-      self = lane == .individual ? .individual(billing) : .team(quantity)
-    } else if let billing = billing {
-      self = .individual(billing)
-    } else if let quantity = quantity {
-      self = .team(quantity)
-    } else {
-      throw unit // FIXME
-    }
-  }
-
-  public func encode(to encoder: Encoder) throws {
-    var container = encoder.container(keyedBy: CodingKeys.self)
-    switch self {
-    case let .individual(billing):
-      try container.encode(billing, forKey: .individual)
-    case let .team(quantity):
-      try container.encode(quantity, forKey: .team)
-    }
+  private enum CodingKeys: String, CodingKey {
+    case lane
+    case billing
+    case quantity
   }
 
   var plan: Stripe.Plan.Id {
-    switch self {
-    case .individual(.monthly):
+    switch (self.billing, self.quantity) {
+    case (.monthly, 1):
       return .individualMonthly
-    case .individual(.yearly):
+    case (.yearly, 1):
       return .individualYearly
-    case .team:
+    case (.monthly, _):
+      return .teamMonthly
+    case (.yearly, _):
       return .teamYearly
     }
   }
 
-  var quantity: Int {
-    switch self {
-    case .individual:
-      return 1
-    case let .team(quantity):
-      return quantity
-    }
-  }
-
-  var billing: Billing {
-    switch self {
-    case let .individual(billing):
-      return billing
-    case .team:
-      return .yearly
-    }
+  var lane: Lane {
+    return self.quantity == 1
+      ? .individual
+      : .team
   }
 
   var isIndividual: Bool {
-    switch self {
-    case .individual:
-      return true
-    case .team:
-      return false
-    }
+    return self.lane == .individual
   }
 
   var isTeam: Bool {
-    switch self {
-    case .team:
-      return true
-    case .individual:
-      return false
+    return self.lane == .team
+  }
+}
+
+extension Pricing: Codable {
```

Comment:
Being a struct means we could auto-derive here, but we do some fancy things on the subscription page with shown/not-shown fields, so this is a bit safer.